script.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop:disable Style/CaseEquality, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      return to_enum unless block_given?
```

Comment:
* [ ]  ```return to_enum(my_each) unless block_given?``` instead of ```return to_enum unless block_given?```
that way
```[1,2,3].my_each``` ```<Enumerator: [1, 2, 3]:my_each>```
```[1,2,3].each``` =>  ```<Enumerator: [1, 2, 3]:each>```

Reply:
method my_each updated